src/braket/circuits/circuit.py

```diff
@@ -762,10 +763,16 @@ def measure(self, target_qubits: QubitSetInput | None = None) -> Circuit:
 
         if target_qubits:
             # Check if the target_qubits are already measured
-            if self._measure_targets and all(
-                target in self._measure_targets for target in target_qubits
+            if (
+                self._measure_targets
+                and any(target in self._measure_targets for target in target_qubits)
```

Comment:
See comment below, but in general. it's slightly more readable to put parentheses around expressions like this to make the precedence obvious